### PR TITLE
Add start to end survey test + fixes

### PR DESCRIPTION
**Pull request by erikh360**

### CODE REVIEW

**Review by rudigiesler — COMMENTED** [review]
Just two minor comments, otherwise looks good!

File: yal/tests/test_segmentation_survey.py [hunk 1]
```diff
@@ -359,6 +360,30 @@ async def test_survey_end(tester: AppTester):
...
+
+        assert (
+            current_question in SURVEY_QUESTIONS[section]["questions"]
+        ), f"Section {section} - broken link to: {current_question}"
+
+        question = SURVEY_QUESTIONS[section]["questions"][current_question]
+
+        paths = []
+        if type(question.get("next")) == dict:
+            paths = set([v for v in question["next"].values()])
```
[the remark this answers is not in the document]
Comment (erikh360) [reply yal/tests/test_segmentation_survey.py:377]: `.values()` returns a `dict_values` type so I added the loop.

File: yal/tests/test_segmentation_survey.py [hunk 2]
```diff
@@ -359,6 +360,30 @@ async def test_survey_end(tester: AppTester):
...
+        else:
+            paths.append(question.get("next"))
+
+        return all([get_next(section, path) for path in paths])
+
+    for section in ["1", "2", "3", "4"]:
```
[the remark this answers is not in the document]
Comment (erikh360) [reply yal/tests/test_segmentation_survey.py:383]: yup

File: yal/tests/test_segmentation_survey.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rudigiesler) [reply yal/tests/test_segmentation_survey.py:377]: Ah okay, but `dict_values` should still be iterable right?
```
>>> d = {"1": "one", "2": "two"}
>>> set(d.values())
{'one', 'two'}
```

**Review by rudigiesler — APPROVED** [review]

**Pull request merged**